doc/ssh_config/ssh_config.md

```diff
@@ -0,0 +1,214 @@
+
+
+###  1.8. <a name='Init flow'></a>Init flow 
+
+During init flow we will set default ssh policies, same as default values in DebianOS. Default values will be added to init_cfg.json.j2, and updated in sshd_config file accordingly.
+####  1.8.1. <a name='FeatureDefault'></a>Feature Default
+
+Description of default values in init_cfg.json regarding ssh server config:
+```
+authentication retries: 6 
+login timeout: 120 //seconds
+ports: 22
+```
+###  1.9. <a name='SAI api'></a>SAI api
+NA
+###  1.10. <a name='Configurationandmanagement'></a>Configuration and management 
+
+####  1.10.1. <a name='SSH_SERVERconfigDBtable'></a>SSH_SERVER configDB table
+
+```
+SSH_SERVER:{
+	POLICIES:{
+		"authentication_retries": {{num}}
+		"login_timeout": {{secs}}
+		"ports": {{num}}
+	}
+}
+```
+#### 1.10.2. ConfigDB schemas
+```
+; Defines schema for SSH_SERVER configuration attributes in SSH_SERVER table:
+key                                   = "POLICIES"             ;ssh server configuration
+; field                               = value
+authentication_retries                = 3*DIGIT                 ; number of login attepmts, should be 100 max
+LOGIN_TIMEOUT                         = 3*DIGIT                 ; login timeout in secs unit, max is 600 secs
+PORTS                                 = 5*DIGIT                 ; ssh port number - max is 65535
+```
+
+####  1.10.3. <a name='CLIYANGmodelEnhancements'></a>CLI/YANG model Enhancements
+```yang
+//filename:  sonic-ssh_server.yang
+module sonic-sshg {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-SSH_SERVER";
+	prefix sshg;
+
+    description "ssh server CONFIG YANG Module for SONiC OS";
+
+	revision 2022-08-29 {
+        description
+            "First Revision";
+    }
+
+    container sonic-SSH_SERVER {
+		container SSH_SERVER {
+			description "ssh server CONFIG part of config_db.json";
+			container POLICIES {
+				leaf authentication_retries {
+					description "number of login attepmts";
+					default 6;
+					type uint8 {
+						range 1..100;
+					}
+				}
+				leaf login_timeout {
+					description "login timeout (secs unit)";
+					default 120;
+					type uint32 {
+						range 1..600;
+					}
+				}
+				leaf ports {
+					description "ssh port numbers";
+					default "22";
+                    type string {
+                        pattern '[0-9]+(,[0-9]+)*' {
+                            error-message "Invalid port numbers value";
+                            error-app-tag ssh-server-ports-invalid-value;
+                        }
+                    }
+				}
+			}/*container POLICIES */
+		} /* container SSH_SERVER  */
+    }/* container sonic-sshg */
+}/* end of module sonic-sshg */
+```
+####  1.10.4. <a name='ConfigDBEnhancements'></a>Config DB Enhancements
+
+The ConfigDB will be extended with next objects:
+
+```json
+{
+	"SSH_SERVER": {
+		"POLICIES":{
+			"authentication_retries": "6",
+			"login_timeout": "120",
+			"ports": "[22]",

```

Comment:
>[ [](http://example.com/codeflow?start=13&length=1)

`[]` is not part of pattern. Is it not aligned with yang model?

Reply:
This comment is still applicable.

Reply:
Our current implementation is using the string as concatenated port values like “22,222,2323”
This methodology is also used in other places such as pfc_enable in sonic-port-qos-map:
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-port-qos-map.yang

Reply:
I understand what you saying. My point is that your sample has a `ports` field with `[]` in value, but there is no `[]` in the pattern in above yang model.

Reply:
This was resolved